java/sage/EPG.java

```diff
@@ -1631,25 +1662,25 @@ public long getCachedProviderIDForName(String s)
   private EPGDataSource currDS;
   private boolean alive = true;
   private java.io.File logoDir;
-  private java.util.Map logoMap;
+  private java.util.Map<String, java.io.File> logoMap;
   //	private java.util.Map colorMap;
 
   private Object stationIDCacheLock = new Object();
   private int[] viewableStationIDCache;
 
-  private java.util.Map lineups;
-  private java.util.Map lineupOverrides;
-  private java.util.Map serviceLevels;
+  private final java.util.Map<Long, java.util.Map<Integer, String[]>> lineups;
+  private final java.util.Map<Long, java.util.Map<Integer, String[]>> lineupOverrides;
+  private final java.util.Map<Long, java.util.Map<Long, Integer>> serviceLevels;
```

Comment:
I made it Integer, but then you had a line that was importing it at long and then I thought that's what it must be.